src/main/proto/protocol.proto

```diff
@@ -119,3 +119,22 @@ message CloseRequest {
 message CloseResponse {
   string error = 1; // err string, empty/omitted if no error
 }
+
+// A SubscriptionOptions object defines the configurable parameters of a STAN Subscription object.
+message SubscriptionOptionsMsg {
```

Comment:
Great idea, but I do not think you should use a protobuf object here.  We are going to someday try to to consolidate protobuf definitions into a file shared across clients or have a process to keep protobuf definitions in sync - this would complicate that process.  Again, good idea, but it may conflict with our plans moving forward.  How about a serializable POJO?